tools/c7n_gcp/tests/test_compute.py

```diff
@@ -11,14 +11,24 @@
 class InstanceTest(BaseTest):
 
     def test_instance_query(self):
-        factory = self.replay_flight_data('instance-query')
+        factory = self.replay_flight_data('instance-query', project_id="cloud-custodian")
         p = self.load_policy(
             {'name': 'all-instances',
              'resource': 'gcp.instance'},
             session_factory=factory)
         resources = p.run()
         self.assertEqual(len(resources), 4)
 
+        self.assertEqual(
+            p.resource_manager.get_urns(resources),
+            [
+                'gcp:compute:us-east1-b:cloud-custodian:instance/custodian-dev',
+                'gcp:compute:us-central1-b:cloud-custodian:instance/c7n-jenkins',
```

Comment:
having zone here feels a little odd in terms of being able to compare or aggregate to region, from an api endpoint perspective it is regional. ie, why zone vs region here?

Reply:
I tried to bring this up on slack too. The problem is around uniqueness in the URN.

For zonal resources it wasn't clear whether there could be a name clash within the same region. Given the self links defined by GCP it seemed to indicate that it may be possible.

If we know that there can be no clash within a region, even across zones, then I'd be all for having just the region there.

The alternative I had for uniqueness was to have the zone as a later element.